src/webgpu/api/validation/encoding/cmds/render/direct_draw.spec.ts

```diff
@@ -0,0 +1,79 @@
+export const description = `
+Here we test the vertex buffer validation for direct draw function. For drawIndexed We only tests
+index buffer and instance step mode vertex buffer OOB. Vertex step mode vertex buffer OOB for
+drawIndexed is covered in robust access.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { ValidationTest } from '../../../validation_test.js';
+
+class F extends ValidationTest {
+  // TODO: Implement the helper functions
+}
+
+export const g = makeTestGroup(F);
+
+g.test(`unused_buffer_bound`)
+  .desc(
+    `
+In this test we test that a small buffer bound to unused buffer slot won't cause validation error.
+- All draw commands,
+  - An unused {index , vertex} buffer with uselessly small range is bound (immediately before draw
+    call)
+`
+  )
+  .unimplemented();
+
+g.test(`index_buffer_OOB`)
+  .desc(
+    `
+In this test we test that index buffer OOB is catched as validation error in drawIndex.
+drawIndexedIndirect didn't has such validation yet.
```

Comment:
> yet

This won't change in future versions (unless we change the spec prior to 1.0).
```suggestion
In this test we test that index buffer OOB is catched as validation error in drawIndexed.
(drawIndexedIndirect doesn't produce a validation error in this case as it is GPU-validated.)
```

Reply:
Done.